raiden-ts/src/helpers.ts

```diff
@@ -341,3 +344,51 @@ export const isValidUrl = (url: string): boolean => {
       : /^(?:(http|https):\/\/)?([^\s\/$.?#&"']+\.)*[^\s\/$?#&"']+(?:(\d+))*$/;
   return regex.test(url);
 };
+
+/**
+ * Construct entire ContractsInfo using UserDeposit contract address as entrypoint
+ *
+ * @param provider - Ethers provider to use to fetch contracts data
+ * @param userDeposit - UserDeposit contract address as entrypoint
+ * @returns contracts info, with blockNumber as block of first registered tokenNetwork
+ */
+export async function fetchContractsInfo(
+  provider: JsonRpcProvider,
+  userDeposit: Address,
+): Promise<ContractsInfo> {
+  const userDepositContract = UserDepositFactory.connect(userDeposit, provider);
+
+  const oneToN = (await userDepositContract.one_to_n_address()) as Address;
```

Comment:
You retrieve this contract pretty early in the function. It gets not used before inserted into the return value. I would think this can make sense if the "order" (also if there is no such forced order) of the return-object has this contrast as first. But since it is not, I would rather retrieve this information more close to where it gets used.

Reply:
Sure, will do